Sage/carbon
# fix(textarea): stop unwanted scroll of container when expandable

When a textarea is `expandable`, it is resized after every render, as well as on resize. This works by first setting the height to 0, then the appropriate size to contain the text. Setting to 0 first is necessary in some scenarios, but when the textarea is in a scrollable container, it makes the container scroll to the top, and then when the height is reset, the browser only scrolls the container down far enough to reach the cursor position - which may be not as far down as where the user was previously. The result is an unwanted and unexpected "jump" of the container scroll position, particularly when typing in a controlled textarea. This is now fixed by taking note of the previous scroll position and resetting it after resizing is complete.

fix #6088

### Proposed behaviour

When a textarea is in a scrollable container, the container should not change its scroll position while the user is editing the textbox



### Current behaviour

Unwanted "jumps" in scroll position happen whenever the window resizes or the component rerenders - most noticeably while the user is editing a controlled textarea.



### Checklist




#### QA


### Additional context



### Testing instructions



As well as checking that the issue is fixed (see Codesandbox linked to the issue), perform regression testing on `expandable` Textareas, both with and without the `rows` prop.

Note: in the original version of this PR, the bug could still be observed (but only once) in situations where the Sage UI font is slow to load - which in particular seems to happen more often than not on CodeSandbox (it happened much more rarely on storybook, although even there I did observe it very occasionally). To observe this, see this old CSB build https://codesandbox.io/s/textarea-focus-zsm8tw and notice how quite often, a few seconds after loading there is a noticeable jump or flicker in the textarea contents as the font loads, and subsequently a scrollbar appears on the textarea itself (not just on the container). After that has happened, when you scroll to the bottom and edit text - even if it's quite a bit later than this "flicker" happened - the original bug of an unwanted jump in scroll position is observed.

Note that disabling the cache in Chrome's developer tools (Network tab) seems to make the bug more likely to happen - but it occurs around 50% of the time even without this.

This issue should be fixed in the latest build - the slow font-loading still happens, and you still see the flicker and even the scrollbar appearing. But now the scrollbar should disappear shortly afterwards, and the original bug should not reappear if you scroll down and enter text later.
